src/components/CurrencyConverter.jsx

```diff
@@ -1,119 +1,269 @@
+    setRightCurrencyData(leftCurrencyData);
   };
 
   const handleConvert = () => {
-    let conversionRate;
-    if (reversed === true) {
-      conversionRate = rightCurrencyValue / leftCurrencyValue;
-    } else {
-      conversionRate = leftCurrencyValue / rightCurrencyValue;
-    }
+    const conversionRate =
+      rightCurrencyData.market_data.current_price[displayCurrency] /
+      leftCurrencyData.market_data.current_price[displayCurrency];
     const result = (inputValue / conversionRate).toFixed(6);
     setConvertedResult(result);
   };
 
+  const getSelectedCurrencyData = async (item) => {
+    try {
+      setSingleCoinIsLoading(true);
+      setSingleCoinLoadingHasError(false);
+      const singleCoinData = await axios(
+        `https://api.coingecko.com/api/v3/coins/${item}?localization=false&tickers=false&market_data=true&community_data=true&developer_data=false&sparkline=false`
+      );
+      setSingleCoinIsLoading(false);
+      if (!singleCoinData) {
+        setSingleCoinLoadingHasError(true);
+        return;
+      } else {
+        if (selectLeftCurrency) setLeftCurrencyData(singleCoinData.data);
+        if (!selectLeftCurrency) setRightCurrencyData(singleCoinData.data);
+      }
+    } catch (err) {
+      setSingleCoinLoadingHasError(true);
+      setSingleCoinIsLoading(false);
+    }
+  };
+
   const handleLeftCurrencySelect = (value) => {
     setLeftCurrency(value);
-    setLeftCurrencyBeforeReverse(value);
-    setLeftCurrencyValue(singleCoin.market_data.current_price[value]);
+    setSelectLeftCurrency(true);
   };
 
+  useEffect(() => {
+    if (selectLeftCurrency) {
+      getSelectedCurrencyData(leftCurrency);
+    } else {
+      getSelectedCurrencyData(rightCurrency);
+    }
+  }, [leftCurrency, rightCurrency]);
+
   const handleRightCurrencySelect = (value) => {
     setRightCurrency(value);
-    setRightCurrencyBeforeReverse(value);
-    setRightCurrencyValue(singleCoin.market_data.current_price[value]);
+    setSelectLeftCurrency(false);
   };
 
   useEffect(() => {
-    getSingleCoinData("bitcoin");
-  }, []);
+    setGetLeftCurrencyPriceVolumeHasError(false);
+    getCoinPriceVolume(leftCurrency, displayCurrency, currencyConverterDays)
+      .then((response) => {
+        if (!response) {
+          setGetLeftCurrencyPriceVolumeHasError(true);
+          return;
+        } else {
+          setLeftCurrencyPriceVolume(response);
+        }
+      })
+      .catch(() => setGetLeftCurrencyPriceVolumeHasError(true));
+  }, [leftCurrency, currencyConverterDays]);
+
+  useEffect(() => {
+    setGetRightCurrencyPriceVolumeHasError(false);
+    getCoinPriceVolume(rightCurrency, displayCurrency, currencyConverterDays)
+      .then((response) => {
+        if (!response) {
+          setGetRightCurrencyPriceVolumeHasError(true);
+          return;
+        } else {
+          setRightCurrencyPriceVolume(response);
+        }
+      })
+      .catch(() => setGetRightCurrencyPriceVolumeHasError(true));
```

Comment:
You're using async/await in the `getSelectedCurrencyData`, and promises here. Try to be consistent in the way you fetch data.